src/esp/physics/CollisionGroupHelper.h

```diff
@@ -0,0 +1,236 @@
+    collisionGroupMasks.at(group) = mask;
+  }
+
+  /**
+   * @brief Get the collision group by its name. Must pass a valid name.
+   *
+   * @param groupName The collision group's configured name.
+   */
+  static CollisionGroup getGroup(const std::string& groupName) {
+    ESP_CHECK(collisionGroupNames.count(groupName) != 0,
+              "Invalid groupName provided. Matches no CollisionGroup.");
+    return collisionGroupNames.at(groupName);
+  }
+
+  /**
+   * @brief Get a collision group's configured name.
+   *
+   * @param group The collision group.
+   * @return The group's configured name.
+   */
+  static std::string getGroupName(const CollisionGroup& group) {
+    for (std::map<std::string, CollisionGroup>::iterator it =
+             collisionGroupNames.begin();
+         it != collisionGroupNames.end(); ++it) {
+      if (group == it->second) {
+        return it->first;
+      }
+    }
+    // enum input, so should not get here unless the map is corrupted
+    CORRADE_INTERNAL_ASSERT_UNREACHABLE();
+    return "";
+  }
+
+  /**
+   * @brief Set a custom name for a collision group.
+   *
+   * Fails if the desired name is already in use.
+   *
+   * @param group The collision group.
+   * @return Whether or not the set was successful.
+   */
+  static bool setGroupName(const CollisionGroup& group,
+                           const std::string& newName) {
+    auto currentName = getGroupName(group);
+    if (collisionGroupNames.count(newName) != 0) {
+      LOG(WARNING) << "CollisionGroupHelper::setGroupName - requested group "
+                      "name is already in use, aborting.";
+      return false;
+    }
+    collisionGroupNames[newName] = collisionGroupNames.at(currentName);
+    collisionGroupNames.erase(currentName);
+    return true;
+  }
+
+  /**
+   * @brief Get a list of all configured collision group names for easy
+   * iteration over groups.
+   *
+   * @return List of configured group names
+   */
+  static std::vector<std::string> getAllGroupNames() {
+    std::vector<std::string> groupNames(collisionGroupNames.size());
+    for (std::map<std::string, CollisionGroup>::iterator it =
+             collisionGroupNames.begin();
+         it != collisionGroupNames.end(); ++it) {
+      groupNames.push_back(it->first);
+    }
+    return groupNames;
+  }
+
+  //! get the mask for group split into boolean indicators of interaction with
+  //! each CollisionGroup
+  /**
+   * @brief Get the collision mask for a group split into a map with entries
+   * corresponding to whether a mask bit is set indicating interaction with each
+   * other group.
+   *
+   * @param group The collision group for which the mask will be queried.
+   * @return The group mask split into per-group booleans indicating interaction
+   * with each group.
+   */
+  static std::map<CollisionGroup, bool> getSplitMask(
+      const CollisionGroup& group) {
+    std::map<CollisionGroup, bool> splitMask;
```

Comment:
Since collision group is an unsigned int, this makes more sense as a https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1BoolVector.html

Reply:
Actually, a bool vector is overkill, it's just it's own unsigned int, right?